src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,198 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> last;
+    private Node<T> first;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(value);
+        if (isEmpty()) {
+            addElementToEmptyList(node);
+        } else {
+            addElementToEndOfTheList(node);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+        Node<T> newNode = new Node<>(value);
+        if (isEmpty()) {
+            addElementToEmptyList(newNode);
+        } else if (size == index) {
+            addElementToEndOfTheList(newNode);
+        } else if (index == 0) {
+            addElementToFirstPosition(newNode);
+        } else {
+            addElementToTheMiddle(newNode, index);
+        }
+        size++;
```

Comment:
should be simplified
also, create `linkTail` method
```suggestion
if (index == 0) {
            linkHead(value);
            return;
        }

        if (index == size) {
            linkTail(value);
            return;
        }
```